rocAL/source/augmentations/effects_augmentations/node_normalize.cpp

```diff
@@ -43,14 +43,42 @@ void NormalizeNode::create_node() {
     int mean_stddev_array_size = 1;
     auto nDim = _inputs[0]->info().num_of_dims() - 1;
     std::vector<unsigned> axis(nDim);
-    for (unsigned i = 0; i < _batch_size; i++) {
-        int totalElements = 1;
-        unsigned *tensor_shape = _inputs[0]->info().roi()[i].end;
-        for (uint j = 0; j < nDim; j++) {
-            axis[j] = ((_axis_mask & (int)(pow(2, j))) >= 1) ? 1 : 0;
-            totalElements *= !axis[j] ? tensor_shape[j] : 1;
+    auto data_layout = _inputs[0]->info().layout();
+    auto tensor_dims = _inputs[0]->info().dims();
+    if (data_layout == RocalTensorlayout::NHWC) {
+        for (unsigned i = 0; i < _batch_size; i++) {
+            int totalElements = 1;
+            unsigned *tensor_shape = _inputs[0]->info().roi()[i].end;
+            axis[0] = ((_axis_mask & (int)(pow(2, 0))) >= 1) ? 1 : 0;
+            totalElements *= !axis[0] ? tensor_shape[1] : 1;
```

Comment:
please change to something like this which is more efficient. 
totalElements *= tensor_shape[1] & (_axis_mask & 0x1);
Also if the tensor_shape are initialized to 1 of axis_mask is 0, the above statement will work for both cases